docs: update velero disaster restoration & backup docs (#826)
elastisys/compliantkubernetes: docs/operator-manual/disaster-recovery.md

velero restore create --from-schedule velero-daily-backup --wait
```

!!!tip

>     Use `velero restore create --help` for velero commands.

Reviewer: ```suggestion
    Use `velero restore create --help` to see available flags and some examples.
```

>     If the last backup is in a phase of PartiallyFailed the argument `--allow-partially-failed` can be used for the existing backup.

Reviewer: ```suggestion
    If a backup has a status of PartiallyFailed, the argument `--allow-partially-failed` can be used to restore from such a backup.
```

>     If a backup or restoration gets stuck or has other issues, look at the [guide](troubleshooting.md#velero-backup-stuck-in-progress).

Reviewer: ```suggestion
    If a backup or restoration gets stuck or has other issues, refer to this [guide](troubleshooting.md#velero-backup-stuck-in-progress).
```

velero restore create --from-schedule velero-daily-backup --wait
```

> !!!tip

Reviewer 2: I think you missed an indentation below:
![image](https://github.com/elastisys/compliantkubernetes/assets/1660833/d0dc4d23-60ea-401b-bb8d-eec5278ecef2)

    Use `velero restore create --help` for velero commands.
    If the last backup is in a phase of PartiallyFailed the argument `--allow-partially-failed` can be used for the existing backup.
    If a backup or restoration gets stuck or has other issues, look at the [guide](troubleshooting.md#velero-backup-stuck-in-progress).

> This command will wait until the restore has finished. You can also do partial restorations, e.g. just restoring one namespace, by using different arguments.

Reviewer 3: Nit: "one sentece per line" rule
```suggestion
This command will wait until the restore has finished.
You can also do partial restorations, e.g. just restoring one namespace, by using different arguments.
```

velero restore create --from-schedule velero-daily-backup --wait
```

!!!tip
Use `velero restore create --help` to see available flags and some examples.
If a backup has a status of PartiallyFailed, the argument `--allow-partially-failed` can be used to restore from such a backup.

> If a backup or restoration gets stuck or has other issues, refer to this [guide](troubleshooting.md#velero-backup-stuck-in-progress).

Reviewer: ```suggestion
If a backup or restore gets stuck or has other issues, refer to this [guide](troubleshooting.md#velero-backup-stuck-in-progress).
```

So a restore will not wipe the volume clean and then restore.
If a clean wipe is the desired behavior, then the volume must be wiped manually before restoring.

### Example scenario

> A Velero backup has failed.
> A pod has not been backed due to the fact it was not ready.
> The backup is in the Phase of "PartiallyFailed".
> This backup can be used by:
> ```bash

> > velero restore create <restore-name> --allow-partially-failed true --from-schedule velero-daily-backup --wait

Reviewer 3: question: Is this really needed?
```suggestion
> velero restore create <restore-name> --allow-partially-failed --from-schedule velero-daily-backup --wait
```

Author: hmmm, dont know why/how i assumed i would need it, was it a old example? Weird :thinking: However, i have now removed it! :+1:

Author: ```
flags:
      --allow-partially-failed optionalBool[=true]      If using --from-schedule, whether to consider PartiallyFailed backups when looking for the most recent one. This flag has no effect if not using --from-schedule.
```

But it seems to default to true :thinking:

So a restore will not wipe the volume clean and then restore.
If a clean wipe is the desired behavior, then the volume must be wiped manually before restoring.

> ### Example scenario

Author: Not really other than trying to give some structure to the text :shrug:

> ### Example scenario

Reviewer: I would prefer if it's not in a blockquote as to follow the same format as the rest of the document

> ### Example scenario

Author: Fixat!

So a restore will not wipe the volume clean and then restore.
If a clean wipe is the desired behavior, then the volume must be wiped manually before restoring.

### Example scenario

A Velero backup has failed.

> A pod has not been backed due to the fact it was not ready.

Reviewer: Q: Are these two lines supposed to indicate why a velero backup has a partially failed status?

Reviewer: Can you try to reformulate so that the intention is clearer :)?

Author: I re-wrote the peace as i felt it was very unclear from the beginning

So a restore will not wipe the volume clean and then restore.
If a clean wipe is the desired behavior, then the volume must be wiped manually before restoring.

### Example

> A Velero backup has failed and a pod has not been backed due to the fact it was not `Ready`.
> The backup is in the Phase of `PartiallyFailed`.
> A backup that has `PartiallyFailed` can be used by the `--allow-partially-failed` flag

Reviewer: What do you think of just this?

```suggestion
A backup that has status `PartiallyFailed` can be restored by using `--allow-partially-failed` flag
```

    ./bin/ck8s ops kubectl wc -n velero delete backupstoragelocations.velero.io backup
    ```

### Troubleshooting

> [Documentation](https://velero.io/docs/v1.13/troubleshooting).

Reviewer 2: I'd like to see all troubleshooting collecting in a single place. Could we point the user here instead: https://elastisys.io/compliantkubernetes/operator-manual/troubleshooting/#velero-backup-stuck-in-progress

We can then point to upstream Velero troubleshooting documentation from our troubleshooting page.
